game-app/game-core/src/main/java/games/strategy/triplea/ui/SimpleUnitPanel.java

```diff
@@ -120,7 +125,13 @@ public void setUnitsFromRepairRuleMap(
    */
   public void setUnitsFromCategories(final Collection<UnitCategory> categories) {
     removeAll();
-    for (final UnitCategory category : categories) {
+    List<UnitCategory> unitCategories = new ArrayList(categories);
+    if (unitCategories.isEmpty()) {
+      return;
+    }
+    final GameData gameData = unitCategories.get(0).getUnitAttachment().getData();
```

Comment:
`UiContext` might have a reference to `GameData` that can be used.
I'm not sure if the game data referenced by unit attachment is always non-null (it might be, in which case this is fine)